File: VMusicPlayerServer/src/main/kotlin/com/github/rain1208/vmusicplayerserver/domain/models/Playlist.kt
package com.github.rain1208.vmusicplayerserver.domain.models

import java.time.LocalDateTime

data class Playlist(
    val playlistId: String,
    val name: String,
    val songCount: Int = 0,
    val firstSongThumbnail: String? = null,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
)

File: VMusicPlayerApp/types/playlist.ts
import type { Song } from './song'

export interface Playlist {
  playlistId: string
  name: string
  songCount: number
  firstSongThumbnail?: string
  createdAt: string
  updatedAt: string
}

export interface PlaylistSongItem {
  position: number
  song: Song
}

export interface PlaylistDetail {
  playlistId: string
  name: string
  songs: PlaylistSongItem[]
  createdAt: string
  updatedAt: string
}

export interface PlayQueue {
  songs: Song[]
  currentIndex: number
}

